client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/protocol/InternalFlag.java

```diff
@@ -0,0 +1,26 @@
+package org.infinispan.client.hotrod.impl.protocol;
+
+/**
+ * Flags which are internal to the protocol and shouldn't be exposed to users.
```

Comment:
I'm OK with making the changes suggested if you guys think that's the way to go. Just trying to justify the two proposals and present another option:

The flag is hidden from the user because he is already specifying the time unit on cache operations. It's a bit confusing to do put(k, v, duration, TimeUnit.MILLISECONDS) only to discover the duration was rounded up to one second because you didn't also set Flag.<MILLIS_FLAG>. It should be set automatically in my opinion. Also I don't see what the user has to gain by having access to this flag given that it controls just the semantic of data on the wire (it's converted to ms on the server anyway) they would just do manually the same thing - set the flag if the duration is not represented exactly by seconds or worse set it to <MILLIS_FLAG> when sending data which can be represented exactly by seconds and waste a bit of bandwidth on each message.

Implementations can completely ignore this change and continue to send SECONDS just like before (except for the vInt -> vLong type change which they will need to implement, this is why I prefered the implementation in the initial PR with new fields, with that approach they can ignore the new fields completly and use the existing fields which continue to have the same type and semantic).

Even if they decide to use ns durations I don't think it's that complicated to check if the duration can be represented exactly in seconds or not.

Regarding not supporting NANOSECONDS: as said previously if the cache API allows you to pass TimeUnit.NANOSECONDS as argument to calls it's strange to limit the HR protocol and not be able to represent exactly the duration specified by the user regardless if at the moment or in the future in distributed mode we can or cannot guarantee that the entry will be expired after that duration on all nodes.

Another approach which I've dismissed because I wanted make as few changes to the protocol as possible is to just send the exact duration specified by the user + the time unit specified. This will require just one conversion on the server but more changes in clients when implementing 2.2.